pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -70,18 +70,29 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 	original, err := r.knativeServingLister.KnativeServings(namespace).Get(name)
 	if apierrs.IsNotFound(err) {
 		// The resource was deleted
-		r.servings.Delete(key)
-		if r.servings.Len() == 0 {
-			r.config.DeleteAll(&metav1.DeleteOptions{})
+		// Report the KnativeServing edit when deleting an existed KnativeServing
+		if _, ok := r.servings[key]; ok {
+			r.StatsReporter.ReportKnativeServingEdit(key)
+
+			delete(r.servings, key)
+			if len(r.servings) == 0 {
+				r.config.DeleteAll(&metav1.DeleteOptions{})
+			}
 		}
 		return nil
-
 	} else if err != nil {
 		r.Logger.Error(err, "Error getting KnativeServing")
 		return err
 	}
-	// Keep track of the number of KnativeServings in the cluster
-	r.servings.Insert(key)
+	// Keep track of the number and generation of KnativeServings in the cluster
+	// Report the KnativeServing edit when creating a new KnativeServing or
+	// changing an existed KnativeServing
+	newGen := original.Generation
+	oldGen, ok := r.servings[key]
+	if (ok && newGen == oldGen+1) || (!ok && newGen == 1) {
+		r.StatsReporter.ReportKnativeServingEdit(key)
```

Comment:
Can the generation be attached?

Reply:
Attaching the generation may be not that useful for the metrics only counting the number of changes.